src/cpuid/src/amd/mod.rs

```diff
@@ -0,0 +1,303 @@
+                .ok_or(NormalizeCpuidError::MissingLeaf0x80000000)?;
+            // SAFETY: Safe, as `0x8000_001f` is within the known range.
+            unsafe {
+                leaf_80000000
+                    .eax
+                    .l_func_ext_mut()
+                    .unchecked_assign(0x8000_001f);
+            }
+        }
+
+        // Updated extended feature fn entry
+        {
+            // set the Topology Extension bit since we use the Extended Cache Topology leaf
+            let leaf_80000001 = self
+                .leaf_mut::<0x80000001>()
+                .ok_or(NormalizeCpuidError::MissingLeaf0x80000001)?;
+            leaf_80000001.ecx.topology_extensions_mut().on();
+        }
+
+        // Update AMD feature entry
+        {
+            // We don't support more then 128 threads right now.
+            // It's safe to put them all on the same processor.
+            let leaf_80000008 = self
+                .leaf_mut::<0x80000008>()
+                .ok_or(FeatureEntryError::MissingLeaf0x80000008)?;
+            // This value allows at most 64 logical threads within a package.
+            // SAFETY: `7` is within the known range and always safe.
+            unsafe {
+                leaf_80000008.ecx.apic_id_size_mut().unchecked_assign(7);
+            }
+            leaf_80000008
+                .ecx
+                .nt_mut()
+                .checked_assign(u32::from(
+                    cpu_count
+                        .checked_sub(1)
+                        .ok_or(FeatureEntryError::NumberOfPhysicalThreadsOverflow)?,
+                ))
+                .map_err(FeatureEntryError::NumberOfPhysicalThreads)?;
+        }
+
+        // Update extended cache topology entry
+        {
+            let leaf_8000001d: Leaf8000001dMut = self.leaf_mut::<0x8000001d>();
+            for subleaf in leaf_8000001d.0 {
+                match u32::from(&subleaf.eax.cache_level()) {
+                    // L1 & L2 Cache
+                    // The L1 & L2 cache is shared by at most 2 hyper-threads
+                    1 | 2 => subleaf
+                        .eax
+                        .num_sharing_cache_mut()
+                        // SAFETY: We know `cpus_per_core > 0` therefore this is always safe.
+                        .checked_assign(u32::from(unsafe {
+                            cpus_per_core.checked_sub(1).unwrap_unchecked()
+                        }))
+                        .map_err(ExtendedCacheTopologyError::NumSharingCache)?,
+                    // L3 Cache
+                    // The L3 cache is shared among all the logical threads
+                    3 => subleaf
+                        .eax
+                        .num_sharing_cache_mut()
+                        .checked_assign(u32::from(
+                            cpu_count
+                                .checked_sub(1)
+                                .ok_or(ExtendedCacheTopologyError::NumSharingCacheOverflow)?,
```

Comment:
Within the scope of normalize we know `cpus_per_core > 0` it cannot equal 0, thus `cpus_per_core.checked_sub(1)` will always return `Ok` therefore it is safe to use `unwrap_unchecked()`.

Within the scope of normalize `cpu_count` can be `0` thus `cpu_count.checked_sub(1)` may return an error, thus it would not be safe to use `unwrap_unchecked`.

We could use `unwrap_unchecked` here if we passed `cpu_count` as `NonZeroU8`, but I am not sure this is correct.